src/init.cpp

```diff
@@ -1388,30 +1388,10 @@ bool AppInit2()
 
     fReindex = GetBoolArg("-reindex", false);
 
-    // Upgrading to 0.8; hard-link the old blknnnn.dat files into /blocks/
+    // Create blocks directory if it doesn't already exist
     boost::filesystem::path blocksDir = GetDataDir() / "blocks";
-    if (!boost::filesystem::exists(blocksDir)) {
+    if (!boost::filesystem::exists(blocksDir))
```

Comment:
nit: you don't need this 'if'.
`create_directories` does not throw an error when the directory exists, it just returns false (https://www.boost.org/doc/libs/1_69_0/libs/filesystem/doc/reference.html#create_directories).

Reply:
Gotcha, thanks! Just being careful.